### PR TITLE
User stories 4 - 7

### PR DESCRIPTION
**Pull request by muydanny**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def create
     if @review.save
       redirect_to "/shelters/#{shelter_id[:id]}"
     else
+      flash.now[:error] = "Please fill out entire form"
```
Comment (muydanny) [inline app/controllers/reviews_controller.rb]: Can edit message if we seem fit

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@ class ApplicationController < ActionController::Base
 
   before_action :get_favorite#, only: [:new, :create]
 
+  add_flash_types :error
```
Comment (perryr16) [inline app/controllers/application_controller.rb:6]: getting dangerous

Comment (muydanny) [inline app/controllers/application_controller.rb:6]: would like to eventually play with the different notifcation ids like :notice or something

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -4,16 +4,34 @@ def new_review
...
   end
 
+  def edit
+    @review = Review.find(params[:review_id])
+  end
+
+  def update
+    review = Review.find(params[:review_id])
+    if review.update(review_params)
+      redirect_to "/shelters/#{shelter_id[:id]}", notice: "Successfully edited review! Thanks dog."
```
Comment (perryr16) [inline app/controllers/reviews_controller.rb:14]: I like the lingo, could we make this two lines of code?

File: app/views/shelters/show.html.erb [hunk 4]
```diff
@@ -16,11 +16,17 @@
...
   <% (@shelter.reviews || []).each do |review| %>
-    <p><%= review.title %></p>
+    <h3><u><%= review.title %></u></h3>
     <p><%= review.content %></p>
-    <p><%= review.rating %></p>
-    <img id="<%= "review-image-#{review.id}" %>" src="<%= review.image %> image" />
+    <p><b>Rating:</b> <%= review.rating %></p>
+    <% if review.image != "" %>
+      <img id="<%= "review-image-#{review.id}" %>" src="<%= review.image %> image" />
+    <% end %>
```
Comment (perryr16) [inline app/views/shelters/show.html.erb:26]: Images werent showing up when I tested the first iteration

File: spec/features/reviews/user_can_delete_review_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  it "I see a link to delete the review" do
+    shelter = create(:shelter)
+    review = create(:review, shelter: shelter)
+
+    visit "/shelters/#{shelter.id}"
+    expect(page).to have_content(shelter.name)
+    expect(page).to have_content(review.title)
+
+    click_link("Delete Review")
+
```
Comment (perryr16) [inline spec/features/reviews/user_can_delete_review_spec.rb:13]: good place to put a `within` as per today's lesson

File: spec/features/reviews/user_can_not_edit_review_with_invalid_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe "When I fail to enter a title, a rating, and/or content in edit review" do
+  it "I see a flash message indicating that I need to fill in form" do
+    shelter = create(:shelter)
+    review = create(:review, shelter: shelter)
+
```
Comment (perryr16) [inline spec/features/reviews/user_can_not_edit_review_with_invalid_spec.rb:7]: so this is how factory bot creates objects

**Pull request merged**